src/main/java/appeng/parts/AEBasePart.java

```diff
@@ -498,4 +495,13 @@ public ISimplifiedBundle getRenderCache() {
     public void setRenderCache(final ISimplifiedBundle renderCache) {
         this.renderCache = renderCache;
     }
+
+    private static int getSide(ForgeDirection direction) {
+        for (int i = 0; i < ForgeDirection.VALID_DIRECTIONS.length; i++) {
```

Comment:
Rather than looping through ForgeDirection, why not check `direction == ForgeDirection.UNKNOWN || direction == null` and return -1 in that case, else return `direction.ordinal()`?